src/js_mixed.hpp

```diff
@@ -0,0 +1,187 @@
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleObjectID : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        return Mixed(Utils::to_object_id(context, value));
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_object_id(context, mixed.get<ObjectId>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleBinary : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        auto owned_binary_data = Utils::to_binary(context, value);
+        return Mixed(owned_binary_data.get());
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_binary(context, mixed.get<BinaryData>());
+    }
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleTimeStamp : public MixedWrapper<Context, Value> {
+    Mixed wrap(Context context, Value const &value) {
+        auto date = Utils::to_date(context, value);
+
+        double milliseconds = Utils::to_number(context, date);
+        int64_t seconds = milliseconds / 1000;
+        int32_t nanoseconds = ((int64_t)milliseconds % 1000) * 1000000;
+        Timestamp ts(seconds, nanoseconds);
+
+        return Mixed(ts);
+    }
+
+    Value unwrap(Context context, Mixed mixed) {
+        return Utils::from_timestamp(context, mixed.get<Timestamp>());
+    }
+};
+
+template <typename JavascriptEngine>
+class TypeMixed {
+   private:
+    using Context = typename JavascriptEngine::Context;
+    using Value = typename JavascriptEngine::Value;
+    using Utils = js::Value<JavascriptEngine>;
+    using Strategy = MixedWrapper<Context, Value> *;
+
+    std::map<DataType, Strategy> strategies = {
+        {type_String, new HandleString<Context, Value, Utils>},
+        {type_Int, new HandleNumber<Context, Value, Utils, Int>},
+        {type_Float, new HandleNumber<Context, Value, Utils, Float>},
+        {type_Double, new HandleNumber<Context, Value, Utils, Double>},
+        {type_Bool, new HandleBoolean<Context, Value, Utils>},
+        {type_Decimal, new HandleDecimal128<Context, Value, Utils>},
+        {type_ObjectId, new HandleObjectID<Context, Value, Utils>},
+        {type_Binary, new HandleBinary<Context, Value, Utils>},
+        {type_Timestamp, new HandleTimeStamp<Context, Value, Utils>},
+    };
+
+    Strategy get_strategy(DataType type) { return strategies[type]; }
+
+    TypeMixed() {}
+
+   public:
+    static TypeMixed &get_instance() {
+        static TypeMixed<JavascriptEngine> instance;
+        return instance;
+    }
+
+    Value wrap(Context context, Mixed mixed) {
+        auto strategy = get_strategy(mixed.get_type());
+
+        if (strategy == nullptr) {
+            throw std::runtime_error(
+                "The " + TypeDeduction::realm_typeof(mixed.get_type()) +
+                " value is not supported for the mixed type.");
+        }
```

Comment:
If this is hit, it would be a bug in the SDK, wouldn't it? There should be no type in `Mixed` that can't be wrapped to `Value`.

Reply:
That's a good observation, but even in that unlikely case it would be nice to fail gracefully.

Reply:
If we did want to fail gracefully, I would suggest rewording the error message to something that makes it clear to the developer that this is a bug on our end and they should file a ticket.